File: fastcampus/01.완전탐색/14888.js
let fs = require('fs');
// let input = fs.readFileSync('/dev/stdin').toString().split('\n').map(el => el.trim());
let input = fs.readFileSync('ex.txt').toString().split('\n').map(el => el.trim());

const n = parseInt(input[0])
const arr = input[1].split(' ').map(el => parseInt(el))
const oper = input[2].split(' ')
const result = []

function rec(n, depth, result){
  if(depth === arr.length){
    result.push(n)
    return
  }
  for(let i = 0; i < oper.length; i ++){
    if(oper[i] > 0){
      oper[i] -= 1
      let newN;
      if(i === 0){
        newN = n + arr[depth]
      } else if (i === 1){
        newN = n - arr[depth]
      } else if (i === 2){
        newN = n * arr[depth]
      } else{
        if(n < 0){
          newN = parseInt(-n / arr[depth])
          newN = -newN
        } else {
          newN = parseInt(n / arr[depth])
        }
      }
      if(newN === -0){
        newN = 0
      }
      rec(newN, depth + 1, result)
      oper[i] += 1
    }
  }
}

rec(arr[0], 1, result)
result.sort((a, b)=> a-b)
console.log(result[result.length-1])
console.log(result[0])

File: fastcampus/01.완전탐색/14888.py
import sys

input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().strip().split()))
oper = list(map(int, input().strip().split()))

def rec(n, depth, result):
    if depth == len(arr):
        result.append(n)
        return

    for i in range(len(oper)):
        if oper[i] != 0:
            oper[i] -= 1
            if i == 0:
                new_n = n + arr[depth]
            elif i == 1:
                new_n = n - arr[depth]
            elif i == 2:
                new_n = n * arr[depth]
            elif i == 3:
                if n < 0:
                    new_n = -n // arr[depth]
                    new_n = -new_n
                else:
                    new_n = n // arr[depth]
            rec(new_n, depth + 1, result)
            oper[i] += 1

result = []
rec(arr[0], 1, result)
result.sort()

print(result[-1])
print(result[0])
